src/frontend-samples/display-styles-sample/DisplayStylesUI.tsx

```diff
@@ -0,0 +1,106 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp, IModelConnection, ScreenViewport, Viewport } from "@bentley/imodeljs-frontend";
+import "common/samples-common.scss";
+import { ControlPane } from "Components/ControlPane/ControlPane";
+import { ReloadableViewport } from "Components/Viewport/ReloadableViewport";
+import * as React from "react";
+import DisplayStylesApp from "./DisplayStylesApp";
+import { DisplayStyle } from "./Styles";
+import { Toggle } from "@bentley/ui-core";
+
+interface DisplayStylesUIState {
+  activePresetIndex: number;
+  merge: boolean;
+  viewport?: Viewport;
+}
+
+interface DisplayStylesUIProps {
+  iModelName: string;
+  iModelSelector: React.ReactNode;
+  displayStyles: DisplayStyle[];
+}
+
+const CUSTOM_STYLE_INDEX = 0;
+
+export default class DisplayStylesUI extends React.Component<DisplayStylesUIProps, DisplayStylesUIState> {
+
+  public state: DisplayStylesUIState = { activePresetIndex: 2, merge: false };
+
+  // Called by the control and will update the active display style.
+  private readonly _onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const index = Number.parseInt(event.target.value, 10);
+    this.setState({ activePresetIndex: index });
+  }
+
+  // Called by the control and updates wether to also apply the Custom display style.
+  private readonly _onToggle = (isOn: boolean) => {
+    this.setState({ merge: isOn });
+  }
+
+  // Will be triggered once when the iModel is loaded.
+  private readonly _onIModelReady = (_iModel: IModelConnection) => {
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp)
+      IModelApp.viewManager.onViewOpen.addOnce((viewport: ScreenViewport) => {
+        this.setState({ viewport });
+      });
+    else
+      this.setState({ viewport: vp });
+  }
+
+  /** A render method called when the state or props are changed. */
+  public componentDidUpdate(_prevProp: DisplayStylesUIProps, prevState: DisplayStylesUIState) {
+    let updateDisplayStyle = false;
+    if (this.state.viewport?.viewportId !== prevState.viewport?.viewportId)
```

Comment:
I suggest testing for undefined viewport and quick exit at the top of this method.  This line will crash on undefined viewport.

Reply:
The ["?." operator (Optional Chaining)](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) will return undefined if the viewport is undefined.  This case handles both a change to the viewport (2 =/= 1) and initialization of the viewport (1 =/= undefined).  Can confirm with chrome debugger.
I did it this way so the controls can be live while the viewport is loading, then the current state of the controls are applied to the viewport when finished loading.  The intent is to minimize downtime if the user is trying create a custom style in styles.ts and is reloading after running updated code.

Reply:
I stand corrected.